### PR TITLE
Cherry pick Tower credential fixes to stable-2.5

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

Cherry pick #36587, #36662, #36908, and #36917 into `stable-2.5`

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
`tower_credential.py`

##### ANSIBLE VERSION

```
2.5
```

### CODE REVIEW

**Comment by ryanpetrello** [comment]
@samdoran I think I'm going to have to introduce one more minor change to `devel`, and we'll need it here as well:

https://github.com/ansible/ansible/pull/36908

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py [hunk 1]
```diff
@@ -213,22 +249,56 @@ def main():
...
+
+            if module.params.get('user'):
+                user_res = tower_cli.get_resource('user')
+                user = user_res.get(username=module.params.get('user'))
+                params['user'] = user['id']
+
+            if module.params.get('team'):
+                team_res = tower_cli.get_resource('team')
+                team = team_res.get(name=module.params.get('team'))
+                params['team'] = team['id']
```
Comment (AlanCoding) [inline lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py:283]: super minor, but the team attribute isn't actually idempotent

(not work fixing)

Comment (ryanpetrello) [reply lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py:283]: I wouldn't hold up this backport for this, personally.

**Comment by ryanpetrello** [comment]
@samdoran this looks correct to me:

```yml
$ cat make_cred.yml
- hosts: all
  tasks:
  - tower_credential:
      name: TESTING-25
      organization: Default
      state: present
      kind: ssh
      username: ryan
      password: secret
      ssh_key_data: "~/.ssh/id_rsa"
      ssh_key_unlock: foo
      tower_config_file: "test.cfg"
```

```shell
$ pip install git+https://github.com/samdoran/ansible.git@cherry-pick/2.5/36587
$ pip install git+https://github.com/ansible/tower-cli.git
$ tower-cli --version
Tower CLI 3.2.1
$ PYTHONPATH=$VIRTUAL_ENV/lib/python2.7/site-packages ansible-playbook -i localhost, -c local make_cred.yml -v
Using /etc/ansible/ansible.cfg as config file

PLAY [all] ***********************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************
ok: [localhost]

TASK [tower_credential] **********************************************************************************************
changed: [localhost] => {"changed": true, "credential": "TESTING-25", "id": 9, "state": "present"}

PLAY RECAP ***********************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0
```

Note that the username, password, SSH key, and keypass are _all_ set:
screen shot 2018-03-01 at 3 58 48 pm

Also works against a Tower 3.1.5 install:
```shell
# Run against a Tower 3.1.5 host
$ pip install git+https://github.com/samdoran/ansible.git@cherry-pick/2.5/36587
$ pip install git+https://github.com/ansible/tower-cli.git@v3.1.8
$ tower-cli --version
Tower CLI 3.1.8
$ PYTHONPATH=$VIRTUAL_ENV/lib/python2.7/site-packages ansible-playbook -i localhost, -c local make_cred.yml -v
Using /etc/ansible/ansible.cfg as config file

PLAY [all] ***********************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [tower_credential] **********************************************************************************************************************************************************************************************************************
changed: [localhost] => {"changed": true, "credential": "TESTING-25", "id": 8, "state": "present"}

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0
```
screen shot 2018-03-01 at 4 00 10 pm

**Comment by ryanpetrello** [comment]
`shipit`

File: lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py (hunk 1 above)
Comment (AlanCoding) [reply lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py:283]: no, it's probably not fixable anyway

**Pull request merged**